Drivers/Periphs/Src/IMU.c

```diff
@@ -36,5 +57,76 @@ BaseType_t IMU_Init(void) {
  * @return pdTRUE if data was successfully fetched from queue, pdFALSE if queue is empty
  */
 BaseType_t IMU_ReadData(IMUData_t *Data) {
-  // TODO: Read data from queue
+  int pTxData = 0;
+  int pRxData_L = 0;
+  int pRxData_H = 0;
+  for(int i = 0; i < readDataReg[0] - 1; i = i + 2){
+    pTxData = readDataReg[i+1];
+    HAL_SPI_TransmitReceive(&hspi1, &pTxData, &pRxData1, 1, NO_POLLING);
```

Comment:
lsm6 supports reading continuously, reading 1 byte at a time makes code more inefficient (and doesn't really take advantage of the hw capabilities of the lsm6)